### PR TITLE
Fix tree keyboard navigation

### PR DESCRIPTION
**Pull request by garygong**
This fixes an issue with the tree keyboard navigation:
- Tab into tree
- Tab out
- Tab back in
- Click a new item
- The item isn't selected

This also has an additional fix for when hitting left on a branch.  If the branch is already collapsed, it should collapse the parent branch and focus the parent node.

Fixes https://github.com/salesforce/design-system-react/issues/1356

### Additional description

---

### Pull Request Review checklist (do not remove)

* [x] `npm run lint:fix` has been run and linting passes.
* [x] Mocha, Jest (Storyshots), and `components/component-docs.json` CI tests pass (`npm test`).
* [x] Tests have been added for new props to prevent regressions in the future. See [readme](https://github.com/salesforce/design-system-react/blob/master/tests/README.md).
* [x] Review the appropriate Storybook stories. Open [http://localhost:9001/](http://localhost:9001/).
* [x] Review tests are passing in the browser. Open [http://localhost:8001/](http://localhost:8001/).
* [ ] Review markup conforms to [SLDS](https://www.lightningdesignsystem.com/) by looking at [DOM snapshot strings](https://facebook.github.io/jest/docs/en/snapshot-testing.html).
* [ ] Add year-first date and commit SHA to `last-slds-markup-review` in `package.json` and push.
* [ ] Request a review of the deployed Heroku app by the Salesforce UX Accessibility Team.
* [ ] Add year-first review date, and commit SHA, `last-accessibility-review`, to `package.json` and push.
* [ ] While the contributor's branch is checked out, run `npm run local-update` within locally cloned [site repo](https://github.com/salesforce-ux/design-system-react-site) to confirm the site will function correctly at the next release.

### CODE REVIEW

File: components/tree/index.jsx [hunk 1]
```diff
@@ -57,7 +58,7 @@ class Tree extends React.Component {
 		// There is no need to render when blurring a node because focus is either:
 		//  - outside of the tree, or
 		//  - focused on another node in the tree, which triggers its own render
-		if (this.state.treeHasFocus && !nextState.treeHasFocus) {
+		if (!nextState.treeHasFocus) {
```
Comment (garygong) [inline components/tree/index.jsx]: it turns out that this is all we need, and before it was causing the issue described in https://github.com/salesforce/design-system-react/issues/1356

File: components/tree/index.jsx [hunk 2]
```diff
@@ -118,6 +119,11 @@ class Tree extends React.Component {
 		this.setState({ treeHasFocus: false });
 	}
 
+	handleExpandClick (event, data) {
```
Comment (garygong) [inline components/tree/index.jsx]: One thing that we need to do is to trigger a re-render as mentioned in the comment above.  An good thing is that now I see the branch loading placeholders, whereas before I didn't see them.  I guess we weren't triggering the re-renders for the before.

File: components/tree/private/branch.jsx [hunk 3]
```diff
@@ -138,6 +138,11 @@ const handleKeyDownLeft = (event, props) => {
 		);
 		const index = nodes.indexOf(props.parent);
 		if (index !== -1) {
+			props.onExpandClick(event, {
```
Comment (garygong) [inline components/tree/private/branch.jsx]: Hitting left on a branch that's already collapsed should collapse the parent branch and focus the parent node, like how it works for collapsing an item.

**Comment by interactivellama** [comment]
Good find on the loading stencils

**Pull request merged**